R/topics.R

```diff
@@ -167,3 +167,56 @@ topic_must <- function(message, topic) {
     x = paste0("Not ", encodeString(topic, quote = "'"))
   ))
 }
+
+content_info <- function(content_entry, index, pkg, section) {
+
+  if (grepl(".*::.*", content_entry)) {
+    names <- strsplit(content_entry, "::")[[1]]
+    if (!rlang::is_installed(names[1])) {
+      abort(
+        sprintf(
+          "%s must be installed if it is mentioned in the reference index.",
+          names[1]
+        )
+      )
+    }
+
+    path <- downlit::href_topic(names[2], names[1])
+    if (is.na(path)) {
+      abort(
+        sprintf(
+          "Could not find an href for topic %s of package %s",
+          names[2], names[1]
+        )
+      )
+    }
+
+    rd_name <- paste0(downlit:::find_rdname(names[1], names[2]), ".Rd")
+    file <- withr::local_tempfile(fileext = ".html")
+    write_lines(capture.output(tools::Rd2HTML(tools::Rd_db(names[1])[[rd_name]])), file)
+    rd <- xml2::read_html(file)
```

Comment:
not sure it's the best way to parse a local Rd, but I felt using XPath rather than regex was more in line with the rest of pkgdown.

Reply:
The point of this is just to get the title of the Rd?

Reply:
yes

Reply:
I've just remembered about the (GPL-licensed, though) printr package, and have adapted its code. Does this look better?